mlir/include/mlir/Dialect/Linalg/Utils/Utils.h

```diff
@@ -38,6 +38,12 @@ namespace linalg {
 // General utilities
 //===----------------------------------------------------------------------===//
 
+// Returns true if `val` represents a zero-filled tensor, per its defining op.
+bool isZeroTensor(Value val);
```

Comment:
This is a fairly generic hook and nothing specific to Linalg, is it?

Reply:
As it has support for tracing a value's definition through `linalg.generic`s, `linalg.fill` and `linalg.copy` it is kind of tied to linalg, I would say. As such I didn't think the code could be moved to StaticValueUtils.cpp (which is pretty much dialect/op agnostic).